pyiron_atomistics/atomistics/structure/factory.py

```diff
@@ -332,3 +335,59 @@ def element(parent_element, new_element_name=None, spin=None, potential_file=Non
                 parent_element=parent_element, new_element=new_element_name
             )
         return periodic_table.element(new_element_name)
+
+    @staticmethod
+    def aimsgb_info(axis, max_sigma):
+        """
+        Provides a list of possible GB structures for a given rotational axis and upto the given maximum sigma value.
+
+        Args:
+            axis : Rotational axis for the GB you want to construct (for example, axis=[1,0,0])
+            max_sigma (int) : The maximum value of sigma upto which you want to consider for your
+            GB (for example, sigma=5)
+
+        Returns:
+            A list of possible GB structures in the format:
+
+            {sigma value: {'theta': [theta value],
+                'plane': the GB planes")
+                'rot_matrix': array([the rotational matrix]),
+                'csl': [array([the csl matrix])]}}
+
+        To construct the grain boundary select a GB plane and sigma value from the list and pass it to the
+        GBBuilder.gb_build() function along with the rotational axis and initial bulk structure.
+        """
+        return GBInformation(axis=axis, max_sigma=max_sigma)
+
+    @staticmethod
+    def aimsgb_build(axis, sigma, plane, initial_struct, to_primitive=False,
+                 delete_layer='0b0t0b0t', add_if_dist=0.0):
+        """
+        Generate a grain boundary structure based on the aimsgb.GrainBoundary module.
+
+        Args:
+            axis : Rotational axis for the GB you want to construct (for example, axis=[1,0,0])
+            sigma (int) : The sigma value of the GB you want to construct (for example, sigma=5)
+            plane: The grain boundary plane of the GB you want to construct (for example, plane=[2,1,0])
+            initial_struct : Initial bulk structure from which you want to construct the GB (a pyiron
+                            structure object).
+            delete_layer : To delete layers of the GB. For example, delete_layer='1b0t1b0t'. The first
+                           4 characters is for first grain and the other 4 is for second grain. b means
+                           bottom layer and t means top layer. Integer represents the number of layers
+                           to be deleted. The first t and second b from the left hand side represents
+                           the layers at the GB interface. Default value is delete_layer='0b0t0b0t', which
+                           means no deletion of layers.
+            add_if_dist : If you want to add extra interface distance, you can specify add_if_dist.
+                           Default value is add_if_dist=0.0
+            to_primitive : To generate primitive or non-primitive GB structure. Default value is
+                            to_primitive=False
+
+        Returns:
+            Grain boundary structure in pyiron structure file format.
```

Comment:
For consistency with t
```suggestion
            :class:`.Atoms`: final grain boundary structure
```